### DISCUSSION TITLE
مشکل وصل شدن به درگاه آسان پرداخت

### DISCUSSION THREAD

**Discussion by mohammad-ashrafi2020** [Sina-Soltani/Parbad#258] [Q&A]
سلام خسته نباشید ، من موقع وصل شدن به درگاه آسان پرداخت به این ارور بر میخورم : BlockSize must be 128 in this implementation.
و اینکه درگاه به من فقط User Name - Password - MarchanId 
داده می خواستم بدونم اون 2 تا پارامتر دیگه برای چی هستن و با چی باید پر بشن ؟ 
Key 
IV

**Comment by Sina-Soltani** [comment]
درود
اون دو پارامتر برای رمزنگاری اطلاعات هستن که بدون اونها اصلا درخواستی نمیشه ارسال کرد

باید با پشتیبانی درگاه صحبت کنید و ازشون بپرسید تا این دو مقدار رو  به شما بدن

**Comment by mohammad-ashrafi2020** [comment]
ممنون بابت پاسخگویی تون یه سوال دیگه داشتم 
-------------
درگاه هایی که پیاده سازی کردین همه بر مبنای Soap هستن
اگر بخوایم درگاه جدیدی به این پگیج اضافه کنیم هم باید با این ساختار اضافه اش کنیم ؟

**Reply by Sina-Soltani** [reply]
خواهش میکنم

خیر.
هر درگاه با توجه به مستنداتی که ارائه شده پیاده سازی شده
درگاه‌هایی که مستندات ارائه شده از جانب اون بانک با تکنولوژی Soap بوده، با Soap هم پیاده‌سازی شده.
و درگاه‌هایی که با Json و Http ارائه شدن هم با همون تکنولوژی منطبق با مستنداتشون ارائه شدن.
اگر به سایر درگاه‌ها نگاهی بکنید این مورد رو خواهید دید

**Comment by mohammad-ashrafi2020** [comment]
ببخشید من 
EncryptionKey و EncryptionVector
رو ازشون گرفتم ولی باز هم جواب نداد و همون ارور رو داد
به نظرتون مشکل از چیه ؟ این ارور جایی رخ میده که داره اطلاعات رو رمزنگاری میکنه

**Reply by mohammad-ashrafi2020** [reply]
سلام مجدد ، متاسفانه این الگوریتم رمزنگاری توی حالت 256 توی 
,net core
ساپورت نمیشه -  این هم لینک اش - الگوریتم دیگه ای هست که ازش استفاده کنم برا رمزنگاری ؟
https://github.com/dotnet/corefx/blob/release/3.1/src/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/RijndaelImplementation.cs#L37

**Comment by Sina-Soltani** [comment]
درود
بررسی میکنم
